### PR TITLE
Fix: Issue #403 - Dynamically updated nominal interest rate 

### PR DESCRIPTION
**Pull request by tarun0**
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by tarun0** [comment]
Here's the screenshot of the changes, for reference. 

screenshot_2

File: mifosng-android/src/main/res/layout/fragment_add_loan.xml [hunk 1]
```diff
@@ -179,18 +166,31 @@
...
+                    android:layout_weight="0.3">
+
+                    <EditText
+                        android:id="@+id/et_nominal_interest_rate"
+                        android:layout_height="wrap_content"
+                        android:layout_width="match_parent"
+                        android:gravity="start"
+                        android:hint="@string/nominal_interest_rate"
+                        android:imeOptions="actionNext"
+                        android:inputType="number"
```
Comment (ishan1604) [inline mifosng-android/src/main/res/layout/fragment_add_loan.xml]: Would this also support decimals out of the box ? or do you have to supply the flag `TYPE_NUMBER_FLAG_DECIMAL`

**Comment by tarun0** [comment]
Thanks @ishan1604 for pointing it out. I'd missed that. Corrected now.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/loanaccount/LoanAccountFragment.java [hunk 2]
```diff
@@ -96,6 +93,9 @@
     @BindView(R.id.et_nominal_interest_rate)
     EditText et_nominal_interest_rate;
 
+    @BindView(R.id.tv_nominal_rate_year_month)
+    TextView tv_nominal_rate_per_year_month;
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/loanaccount/LoanAccountFragment.java]: change field name to ```tvNominalRatePerYearMonth``` and refactor same as to others

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/loanaccount/LoanAccountFragment.java [hunk 3]
```diff
@@ -509,4 +509,9 @@ public void onItemSelected(AdapterView<?> parent, View view, int position, long
     public void onNothingSelected(AdapterView<?> parent) {
 
     }
+
+    @Override
+    public void setDefaultValues(String nominalRatePerYearMonth) {
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/loanaccount/LoanAccountFragment.java]: change method name to ```showDefaultValues(String nominalRatePerYearMonth);```

Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/loanaccount/LoanAccountFragment.java]: change method name to ```showDefaultValues(String nominalRatePerYearMonth);```

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/loanaccount/LoanAccountPresenter.java [hunk 4]
```diff
@@ -89,6 +89,8 @@ public void onError(Throwable e) {
                     public void onNext(LoanTemplate loanTemplate) {
                         getMvpView().showProgressbar(false);
                         getMvpView().showLoanAccountTemplate(loanTemplate);
+                        getMvpView().setDefaultValues(
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/loanaccount/LoanAccountPresenter.java]: do not call this method here, Call in method ```showLoanAccountTemplate()``` in Fragment

**Comment by tarun0** [comment]
@therajanmaurya Should we follow the `camelCase` or the convention defined here - https://github.com/openMF/android-client/wiki/Code-style-conventions#51-id-naming , which looks like `snake_case` or the mix of the two, like `tv_thisView`?

**Comment by tarun0** [comment]
Made the required changes.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/loanaccount/LoanAccountFragment.java [hunk 5]
```diff
@@ -199,20 +199,20 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle
...
+        disbursementon_date = tvDisbursementOnDate.getText().toString();
+        submittion_date = tvSubmittedOnDate.getText().toString();
         submittion_date = DateHelper.getDateAsStringUsedForCollectionSheetPayload
                 (submittion_date).replace("-", " ");
         disbursementon_date = DateHelper.getDateAsStringUsedForCollectionSheetPayload
                 (disbursementon_date).replace("-", " ");
 
 
-        bt_loan_submit.setOnClickListener(new View.OnClickListener() {
+        btLoanSubmit.setOnClickListener(new View.OnClickListener() {
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/loanaccount/LoanAccountFragment.java]: We are using ButterKnife. Please remove this OnClickListener implementation and use Butter Knife ```@OnClick``` annotation.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/loanaccount/LoanAccountFragment.java [hunk 6]
```diff
@@ -337,9 +337,9 @@ private void initiateLoanCreation(LoansPayload loansPayload) {
     public void inflatesubmissionDate() {
         mfDatePicker = MFDatePicker.newInsance(this);
 
-        tv_submittedon_date.setText(MFDatePicker.getDatePickedAsString());
+        tvSubmittedOnDate.setText(MFDatePicker.getDatePickedAsString());
 
-        tv_submittedon_date.setOnClickListener(new View.OnClickListener() {
+        tvSubmittedOnDate.setOnClickListener(new View.OnClickListener() {
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/loanaccount/LoanAccountFragment.java]: We are using ButterKnife. Please remove this OnClickListener implementation and use Butter Knife ```@OnClick``` annotation.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/loanaccount/LoanAccountFragment.java [hunk 7]
```diff
@@ -352,9 +352,9 @@ public void onClick(View view) {
     public void inflatedisbusmentDate() {
         mfDatePicker = MFDatePicker.newInsance(this);
 
-        tv_disbursementon_date.setText(MFDatePicker.getDatePickedAsString());
+        tvDisbursementOnDate.setText(MFDatePicker.getDatePickedAsString());
 
-        tv_disbursementon_date.setOnClickListener(new View.OnClickListener() {
+        tvDisbursementOnDate.setOnClickListener(new View.OnClickListener() {
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/loanaccount/LoanAccountFragment.java]: We are using ButterKnife. Please remove this OnClickListener implementation and use Butter Knife ```@OnClick``` annotation.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/loanaccount/LoanAccountFragment.java [hunk 8]
```diff
@@ -64,64 +64,64 @@
...
-    CheckBox ck_calculateinterest;
+    CheckBox ckCalculateInterest;
 
     @BindView(R.id.disbursementon_date)
-    TextView tv_disbursementon_date;
+    TextView tvDisbursementOnDate;
 
     @BindView(R.id.bt_loan_submit)
-    Button bt_loan_submit;
+    Button btLoanSubmit;
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/loanaccount/LoanAccountFragment.java]: change to ```bt --> btn```

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/loanaccount/LoanAccountFragment.java [hunk 9]
```diff
@@ -64,64 +64,64 @@
...
-    Spinner sp_loan_officer;
+    Spinner spLoanOfficer;
 
     @BindView(R.id.sp_fund)
-    Spinner sp_fund;
+    Spinner spFund;
 
     @BindView(R.id.ck_calculateinterest)
-    CheckBox ck_calculateinterest;
+    CheckBox ckCalculateInterest;
```
Comment (therajanmaurya) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/loanaccount/LoanAccountFragment.java]: change to ```ckCalculateInterest --> cbCalculateInterest```

**Comment by tarun0** [comment]
@therajanmaurya Made the changes.

**Comment by therajanmaurya** [comment]
@ishan1604 Please give your final review and merge.

**Pull request merged**